pkg/helm/actions/upgrade_release.go

```diff
@@ -45,5 +53,15 @@ func UpgradeRelease(ns, name, url string, vals map[string]interface{}, conf *act
 		}
 	}
 
+	// Ensure chart URL is properly set in the upgrade chart
+	if url != "" {
+		if ch.Metadata.Annotations == nil {
+			ch.Metadata.Annotations = make(map[string]string)
+		}
+		if _, ok := ch.Metadata.Annotations["chart_url"]; !ok {
```

Comment:
hum... in this case you do want to override what is there. This will cover the case where you are upgrading to a different version which will have a diff url. So... I would not check if there is not value to set url. Set it every time.

Reply:
Thanks for pointing this out. Was wondering about that case but went with safe approach, which is sitcking with the first set URL. Will update.

Reply:
Updated.